src/components/transformation/create/Schema/Header.tsx

```diff
@@ -0,0 +1,42 @@
+import { Typography, useTheme } from '@mui/material';
+import { useEditorStore_currentCatalog } from 'components/editor/Store/hooks';
+import { WarningCircle } from 'iconoir-react';
+import { useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+function DerivationSchemaHeader() {
+    const theme = useTheme();
+
+    const currentCatalog = useEditorStore_currentCatalog();
+
+    const schemaUnedited = useMemo(() => {
+        if (currentCatalog && Object.hasOwn(currentCatalog.spec, 'schema')) {
+            return Object.hasOwn(
+                currentCatalog.spec.schema.properties,
+                'your_key'
+            );
+        } else {
+            return false;
+        }
+    }, [currentCatalog]);
```

Comment:
I am not sure exactly how I feel about this being in the head itself. Are you thinking this validation would eventually move somewhere else or stay here?

Reply:
It should be in the store because the primary CTAs need to be disabled when the flag is true. Moving it now.